### DISCUSSION TITLE
Field Validation problems

### DISCUSSION THREAD

**Discussion by InterFelix** [directus/directus#13461] [Archive]
Why doesn't this work?
![grafik](https://user-images.githubusercontent.com/29462559/169620392-ab69d333-4cf6-491b-a78a-447be1e4a62f.png)

Both `Building` and `Room` contain foreign keys referencing `Tribe` (Rooms as well as Buildings belong to a Tribe).
`Room.Building` is a Many-to-One-Relationship from `Room` to `Building` (a Room is in a building).
I now want to validate that:

1.  the user can only create Rooms where `Room.Tribe` equals the tribe referenced in their user object
2.  the user can only create Rooms which reference buildings where `Building.Tribe` equals the tribe referenced in their user object.

I have created an item in `Tribe` which is referenced in an item in `Building`.
If I now try to create an item in `Room` referencing this item in `Building` as well as the same item in `Tribe` the reference item in `Building` also references, I get a validation error message:
![grafik](https://user-images.githubusercontent.com/29462559/169621323-0e0e148d-4003-4b2b-9cfe-e2088e80b971.png)

Also, this would be a great use case for validation rules that allow comparing the values of different fields in the same item (or more specifically fields against sub-fields), like `Room.Tribe **equals** Room.Building.Tribe`.

**Comment by benhaynes** [comment]
Interesting! I believe I see the confusion. The end group is meant to go at the top, with all the conditions nested within. Be sure to drag the conditions to the right a bit so they nest properly. Give that a try and let us know if things work as expected.

**Reply by InterFelix** [reply]
I've tried this, but had no luck:
![grafik](https://user-images.githubusercontent.com/29462559/169647822-2efec650-1738-46fc-9fe3-1e71f3d7b4eb.png)

I still get the same error when trying to create a Room item like before:
![grafik](https://user-images.githubusercontent.com/29462559/169647760-35d8597e-5b8b-4b1b-9891-06b9d2438989.png)

**Reply by InterFelix** [reply]
In fact, to rule out the AND-Group as a potential source of error, I removed it entirely and only included the validation rule for `Building`:
![grafik](https://user-images.githubusercontent.com/29462559/169647922-c704fa71-6856-4ab2-9cf9-293ef7081c78.png)

Still, the symptoms are the same.
It appears there is some problem with validation rules that target "Sub-Fields".

**Reply by InterFelix** [reply]
To reproduce, try the following:

1. Create a collection `Group`
2. Create a custom field "group" in `Directus Users` referencing `Group` through a Many-To-One-Relationship
3. Create a Collection `Building` referencing `Group` through a M2O-Relationship
4. Create a Collection `Room` referencing both `Building` and `Group` through a M2O-Relationship
5. Create a role with a create validation rule for `Room`, validating that `Room.Building.Group` equals `$CURRENT_USER.group.id` and that `Room.Group` also equals `$CURRENT_USER.group.id`
7. Create an item in `Group` and an item in `Building` referencing this item in `Group`
8. Create a user referencing this item in `Group` through their `group` field with the previously created role
9. Try creating an item in `Room` referencing both the item in `Building` and the item in `Group`. 
This will result in a validation error for `Room.Building`, but not for `Room.Group`. The validation rule for `Room.Group` works fine, the one for `Room.Building` does not.
The validation will fail, even though both Room.Building.Group and the users G

**Comment by azrikahar** [comment]
This should be the same issue as #13198 👍 updated the title to reflect the issue clearer.
